src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -307,20 +307,26 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
         HostingEventSource.Log.RequestStart(httpContext.Request.Method, httpContext.Request.Path);
     }
 
-    [MethodImpl(MethodImplOptions.NoInlining)]
-    private Activity? StartActivity(HttpContext httpContext, bool loggingEnabled, bool diagnosticListenerActivityCreationEnabled, out bool hasDiagnosticListener)
+    private static bool TryParseActivityContext(string traceParent, string traceState, bool isRemote, out ActivityContext context)
```

Comment:
Nit: isRemote will always be true for you. You could eliminate the argument and simplify the logic.